### PR TITLE
feat: TinyMCE plugin insert iframe

### PR DESCRIPTION
**Pull request by johnvente**
### Description
This PR adds a TinyMCE plugin that enables the insertion of an iframe with its respective width and height dimensions, along with additional properties. It is similar to the official plugin found at:
[Page Embed Plugin - Tiny Documentation](https://www.tiny.cloud/docs/plugins/premium/pageembed/#usingpageembed)

The plugin contains two tabs:

- **General tab**: This tab allows us to add:
   - Source URL: the URL of the iframe (required)
   - Size: This allows us to add the size of the iframe in pixels
      - Inline value: Allows us to add custom width and height (px format)
      - Big embed: Will add a value of `800px` for the width and the height
      - Small embed: Will add a value of `100px` for the width and the height

- **Advanced tab**: This tab allows us to add metadata (optional) for the iframe
   - Name: name attribute for the iframe
   - Title: title attribute for the iframe
   - Long description URL: longdesc attribute for the iframe
   - Show iframe border: A black border for the iframe container (checkbox)
   - Enable scrollbar: enable scrollbar for the iframe (checkbox)

                       

#### DEMO

https://github.com/openedx/frontend-lib-content-components/assets/134975835/2ddaa4b5-fea1-4bd2-88ac-5be93745dcd9

button-iframe

General tab             |  Advanced tab
:-------------------------:|:-------------------------:
  |  

### How to test it (Devstack)

You need to have this Waffle flag activated

`new_core_editors.use_new_text_editor `

[Doc here](https://github.com/openedx/frontend-app-course-authoring#requirements-1)

1. Go to `~/workspace/frontend-app-course-authoring `
```
 cd ~/workspace/frontend-app-course-authoring 
```

2. Create a file called `module.config.js` with the following config

```
module.exports = {
  localModules: [
    {
      moduleName: "@edx/frontend-lib-content-components/dist",
      dir: "./frontend-lib-content-components",
      dist: "src",
    },
    {
      moduleName: "@edx/frontend-lib-content-components",
      dir: "./frontend-lib-content-components",
      dist: "src",
    },
  ],
};
```
3. Clone frontend-lib-content-components

```
git fetch origin pull/427/head:jv/tiny-mce-plugin-iframe-embed
git checkout jv/tiny-mce-plugin-iframe-embed
```

4. initialize frontend-app-course-authoring 

```
npm start
```

**Note:** Due to jsdom is not supported by tinyMCE we are adding test only for the plugin
check more [here](https://github.com/openedx/frontend-lib-content-components/blob/main/src/editors/sharedComponents/TinyMceWidget/index.test.jsx#L11)

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @johnvente! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

**Comment by johnvente** [comment]
@e0d Hi there! this is ready for review now, I've fixed the failures

**Comment by santiagosuarezedunext** [comment]
Thanks @johnvente for this implementation! Please add unit tests to ensure the robustness of the code.

**Comment by santiagosuarezedunext** [comment]
Hey @jmakowski1123, any suggestions on who might be a good fit to review this? 👍

**Comment by johnvente** [comment]
> Thanks @johnvente for this implementation! Please add unit tests to ensure the robustness of the code.

Hi @santiagosuarezedunext! I've added the minimal test for the plugin

**Comment by johnvente** [comment]
Hi @santiagosuarezedunext! I've added the unit tests for the plugin

File: src/editors/sharedComponents/TinyMceWidget/customTinyMcePlugins/embedIframePlugin.test.js
Comment (mariajgrimaldi) [inline src/editors/sharedComponents/TinyMceWidget/customTinyMcePlugins/embedIframePlugin.test.js:164]: can we specify that we're only testing the iframe plugin?

**Comment by jmakowski1123** [comment]
@mphilbrick211 This one already went through Product Review :) https://openedx.atlassian.net/wiki/spaces/OEPM/pages/3887071233/Problem+Authors+don+t+have+enough+options+to+create+the+content+they+would+like

Let's touch base on how to connect the updated product review process, as this is the preferred way forward (ie, proposals get product review and approval before the PR is submitted, which is exactly what happened here.)

**Comment by santiagosuarezedunext** [comment]
Hi @mphilbrick211 😊
Who do you think would be the right person to review this PR?

**Comment by mphilbrick211** [comment]
> Hi @mphilbrick211 😊 Who do you think would be the right person to review this PR?

Hi @santiagosuarezedunext! The reviewing team would be @openedx/teaching-and-learning, but they are out for the holidays until January.

**Comment by santiagosuarezedunext** [comment]
Hi @mphilbrick211 
I hope you had a fantastic holiday break. I wanted to check in about the reviewing team, @openedx/teaching-and-learning, being out until January. Is there any update or progress on the matter?

Your assistance is greatly appreciated. Thank you!

Best regards.

**Comment by mphilbrick211** [comment]
Thanks for flagging, @santiagosuarezedunext! @openedx/2u-tnl @openedx/teaching-and-learning flagging for you!

File: src/editors/sharedComponents/TinyMceWidget/customTinyMcePlugins/embedIframePlugin.test.js
Comment (johnvente) [reply src/editors/sharedComponents/TinyMceWidget/customTinyMcePlugins/embedIframePlugin.test.js:164]: Added

**Comment by johnvente** [comment]
Hi there @santiagosuarezedunext! Do you think this implementation should it be in a waffle flag?

**Review by mariajgrimaldi — APPROVED** [review]
I tested this in my local environment using Tutor, and it works as described in the cover letter! Thank you. Let me know if anything else changes.

**Comment by johnvente** [comment]
Hi there @navinkarkera! We know that you are not maintainer from this repository but could you take a look in this PR? That would help us a lot. Thank you very much!

**Comment by santiagosuarezedunext** [comment]
@johnvente, it's not necessary. Unlike other cases, this development received product approval without being flagged behind a waffle flag, and this is because it's an isolated improvement that doesn't affect any workflow or add anything new; it's just an enhancement to something that could already be done.
CC: @jmakowski1123 @mphilbrick211

**Review by 01001110J — APPROVED** [review]

**Review by dcoa — APPROVED** [review]

**Review by BryanttV — APPROVED** [review]

**Comment by BryanttV** [comment]
Hi @johnvente, I also tested the implementation using tutor and it worked perfectly!

**Review by brian-smith-tcril — APPROVED** [review]

**Pull request merged**
